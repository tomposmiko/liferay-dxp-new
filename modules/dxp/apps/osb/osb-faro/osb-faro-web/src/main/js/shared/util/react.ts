import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
