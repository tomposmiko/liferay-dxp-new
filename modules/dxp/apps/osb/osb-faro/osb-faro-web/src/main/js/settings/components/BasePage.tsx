import Breadcrumbs from 'shared/components/Breadcrumbs';
import DocumentTitle from 'shared/components/DocumentTitle';
import getCN from 'classnames';
import MaintenanceAlert from 'shared/components/MaintenanceAlert';
import NotificationAlertList, {
	useNotificationsAPI
} from 'shared/components/NotificationAlertList';
import React from 'react';
import TextTruncate from 'shared/components/TextTruncate';
import {connect, ConnectedProps} from 'react-redux';
import {IBreadcrumbArgs} from 'shared/util/breadcrumbs';
import {PageActions} from 'shared/components/base-page/Header';

const connector = connect((_, {children}: {children: React.ReactNode}) => ({
	passedChildren: children
}));

type PropsFromRedux = ConnectedProps<typeof connector>;

interface ISettingsBasePageProps extends PropsFromRedux {
	breadcrumbItems?: Array<IBreadcrumbArgs>;
	className?: string;
	documentTitle?: string;
	groupId: string;
	pageActions?: Array<any>;
	pageActionsDisplayLimit?: number;
	pageDescription?: React.ReactNode;
	pageTitle?: React.ReactNode;
	subTitle?: React.ReactNode;
}

const SettingsBasePage: React.FC<ISettingsBasePageProps> = ({
	breadcrumbItems,
	className,
	documentTitle,
	groupId,
	pageActions = [],
	pageActionsDisplayLimit,
	pageDescription,
	pageTitle,
	passedChildren,
	subTitle
}) => {
	const notificationResponse = useNotificationsAPI(groupId);

	return (
		<div className={getCN('settings-base-page-root', className)}>
			<DocumentTitle
				title={`${documentTitle || pageTitle} - ${Liferay.Language.get(
					'settings'
				)}`}
			/>

			<NotificationAlertList
				{...notificationResponse}
				groupId={groupId}
			/>

			<MaintenanceAlert />

			{breadcrumbItems && <Breadcrumbs items={breadcrumbItems} />}

			{(!!pageTitle || !!pageDescription || !!pageActions.length) && (
				<div
					className={getCN('content-header', {
						['has-page-actions']: !!pageActions.length
					})}
				>
					<div className='header-text'>
						{pageTitle && (
							<div className='d-flex'>
								<h3 className='title-text text-truncate'>
									<TextTruncate title={pageTitle} />
								</h3>

								{subTitle && (
									<TextTruncate
										className='subtitle-text ml-2'
										title={subTitle}
									/>
								)}
							</div>
						)}

						{pageDescription && (
							<div className='description'>{pageDescription}</div>
						)}
					</div>

					<div className='page-actions-container'>
						<PageActions
							actions={pageActions}
							actionsDisplayLimit={pageActionsDisplayLimit}
						/>
					</div>
				</div>
			)}

			<div>{passedChildren}</div>
		</div>
	);
};

export default connector(SettingsBasePage);
