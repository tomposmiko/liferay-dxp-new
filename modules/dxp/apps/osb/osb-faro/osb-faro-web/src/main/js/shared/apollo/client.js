import cache from './cache';
import Uri from 'metal-uri';
import {ApolloClient} from 'apollo-client';
import {
	CommerceAverageOrderValueResolver,
	CommerceAverageRevenuePerAccountResolver,
	CommerceIncompleteOrdersResolver,
	CommerceTotalOrderValueResolver,
	CustomAssetsListResolver,
	EventAnalysisListResolver,
	ExperimentResolver as Experiment
} from './resolvers';
import {get} from 'lodash';
import {HttpLink} from 'apollo-link-http';
import {onError} from 'apollo-link-error';
import {reloadPage} from 'shared/util/router';

const groupIdRegex = /^\/workspace\/([a-z0-9._-]+)/;

/**
 * Fecth With Project ID
 * @param {string} uri
 * @param {object} options
 */
const fetchWithGroupId = (uri, options) => {
	const currentUri = new Uri(window.location.href);
	const {operationName} = JSON.parse(options.body);
	const pathname = currentUri.getPathname();
	const matches = pathname.match(groupIdRegex);

	if (matches !== null && matches.length > 1) {
		const groupId = matches[1];

		return fetch(
			`${uri}?opname=${operationName}&projectGroupId=${groupId}`,
			options
		);
	}

	return fetch(`${uri}/&opname=${operationName}`, options);
};

const client = new ApolloClient({
	addTypename: true,
	cache,
	defaultOptions: {
		watchQuery: {
			notifyOnNetworkStatusChange: true
		}
	},
	link: HttpLink.from([
		onError(({operation}) => {
			const status = get(operation.getContext(), ['response', 'status']);

			if (status === 401) {
				reloadPage();
			}
		}),
		new HttpLink({
			credentials: 'same-origin',
			fetch: fetchWithGroupId,
			uri: '/o/cerebro/graphql'
		})
	]),
	resolvers: {
		Experiment,
		Query: {
			dashboards(_, params) {
				return CustomAssetsListResolver(params);
			},
			eventAnalysisList(_, params) {
				return EventAnalysisListResolver(params);
			},
			orderAccountAverageCurrencyValues(_, params) {
				return CommerceAverageRevenuePerAccountResolver(params);
			},
			orderAverageCurrencyValues(_, params) {
				return CommerceAverageOrderValueResolver(params);
			},
			orderIncompleteCurrencyValues(_, params) {
				return CommerceIncompleteOrdersResolver(params);
			},
			orderTotalCurrencyValues(_, params) {
				return CommerceTotalOrderValueResolver(params);
			}
		}
	}
});

export default client;
