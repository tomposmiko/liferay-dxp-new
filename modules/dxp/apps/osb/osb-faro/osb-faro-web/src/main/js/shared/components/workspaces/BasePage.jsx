import * as API from 'shared/api';
import ClayButton from '@clayui/button';
import ClayIcon from '@clayui/icon';
import DocumentTitle from 'shared/components/DocumentTitle';
import getCN from 'classnames';
import React, {createContext} from 'react';
import UserDropdown from 'shared/components/user-dropdown';
import withCurrentUser from 'shared/hoc/WithCurrentUser';
import {Align} from '@clayui/drop-down';
import {LANGUAGES} from 'shared/util/constants';
import {PropTypes} from 'prop-types';
import {Routes} from 'shared/util/router';

export const BasePageContext = createContext({currentUser: {}});

export class WorkspacesBasePage extends React.Component {
	static defaultProps = {
		backLabel: Liferay.Language.get('back'),
		details: ''
	};

	static propTypes = {
		backLabel: PropTypes.string,
		backUrl: PropTypes.string,
		details: PropTypes.oneOfType([
			PropTypes.array,
			PropTypes.string,
			PropTypes.func
		]),
		title: PropTypes.string
	};

	getUserMenuItems() {
		const {
			currentUser: {emailAddress, languageId, screenName}
		} = this.props;

		return {
			base: [
				{
					items: [
						{
							childMenuId: 'language',
							divider: true,
							label: Liferay.Language.get('language')
						},
						{
							externalLink: true,
							label: Liferay.Language.get('account'),
							url: `https://web.liferay.com/web/${screenName}/account-settings`
						},
						{
							externalLink: true,
							label: Liferay.Language.get('sign-out'),
							url: Routes.LOGOUT
						}
					],
					subheaderLabel: emailAddress
				}
			],
			language: [
				{
					items: LANGUAGES.map(({id, label}) => {
						const active = languageId === id;

						return {
							active,
							label,
							onClick: active
								? null
								: () =>
										API.user
											.updateLanguage({
												languageId: id
											})
											.then(() =>
												window.location.reload()
											)
						};
					})
				}
			]
		};
	}

	render() {
		const {
			backLabel,
			backURL,
			children,
			className,
			currentUser,
			details,
			title
		} = this.props;

		return (
			<BasePageContext.Provider value={{currentUser}}>
				<div className={getCN('workspaces-base-page-root', className)}>
					<DocumentTitle title={title} />

					<div className='header-container'>
						<a href='https://liferay.com' target='_blank'>
							<ClayIcon
								className='icon-root liferay-logo'
								symbol='liferay-logo'
							/>
						</a>

						<UserDropdown
							alignmentPosition={Align.BottomRight}
							initialActiveMenu='base'
							menus={this.getUserMenuItems()}
							showCaret
							userName={currentUser.name}
						/>
					</div>

					<div className='content'>
						<div className='content-container'>
							{backURL && (
								<div className='back-container'>
									<ClayButton
										className='button-root'
										displayType='unstyled'
										href={backURL}
									>
										<ClayIcon
											className='icon-root'
											symbol='angle-left'
										/>

										{backLabel}
									</ClayButton>
								</div>
							)}

							<div className='title-container'>
								<div className='logo-container'>
									<ClayIcon
										className='icon-root logo-icon'
										symbol='ac-logo'
									/>

									<span className='logo-text'>
										{Liferay.Language.get(
											'analytics-cloud'
										)}
									</span>
								</div>

								<h1 className='title'>{title}</h1>

								<div className='details-container'>
									{details}
								</div>
							</div>

							{children}
						</div>
					</div>
				</div>
			</BasePageContext.Provider>
		);
	}
}

export default withCurrentUser(WorkspacesBasePage);
