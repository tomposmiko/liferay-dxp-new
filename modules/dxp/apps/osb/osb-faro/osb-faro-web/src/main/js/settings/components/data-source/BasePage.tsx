import BasePage from 'settings/components/BasePage';
import DataSourceStatus from './DataSourceStatus';
import getCN from 'classnames';
import React from 'react';
import {addAlert} from 'shared/actions/alerts';
import {Alert, Modal} from 'shared/types';
import {close, modalTypes, open} from 'shared/actions/modals';
import {compose} from 'redux';
import {connect} from 'react-redux';
import {DataSource, User} from 'shared/util/records';
import {deleteDataSource} from 'shared/actions/data-sources';
import {
	getDataSourceDisplayObject,
	hasLegacyDXPConnection
} from 'shared/util/data-sources';
import {Routes, toRoute} from 'shared/util/router';
import {sub} from 'shared/util/lang';
import {truncate} from 'lodash';
import {withHistory} from 'shared/hoc';

const getPageDescription = dataSource =>
	dataSource
		? [dataSource.name, dataSource.url]
				.filter(item => item)
				.map(item => truncate(item, {length: 50}))
				.join(' - ')
		: '';

interface IBaseDataSourcePageProps extends React.HTMLAttributes<HTMLElement> {
	addAlert: Alert.AddAlert;
	close: Modal.close;
	currentUser: User;
	documentTitle: string;
	dataSource: DataSource;
	deleteDataSource: ({
		groupId,
		id
	}: {
		groupId: string;
		id: string;
	}) => Promise<void>;
	groupId: string;
	history: {
		push: (path: string) => void;
	};
	id: string;
	open: Modal.open;
	pageDescription: string;
	pageTitle: React.ReactNode;
	passedChildren: React.ReactNode;
	showDelete: boolean;
}

const BaseDataSourcePage: React.FC<IBaseDataSourcePageProps> = ({
	addAlert,
	className = '',
	close,
	currentUser,
	dataSource,
	deleteDataSource,
	documentTitle = Liferay.Language.get('configure-data-source'),
	groupId,
	history,
	id,
	open,
	pageDescription,
	pageTitle = Liferay.Language.get('configure-data-source'),
	passedChildren,
	showDelete = false,
	...otherProps
}) => {
	const name = dataSource ? dataSource.name : '';

	const handleDeleteClick = () => {
		open(modalTypes.DELETE_CONFIRMATION_MODAL, {
			children: (
				<p>
					<strong>
						{sub(
							Liferay.Language.get(
								'to-delete-x,-copy-the-sentence-below-to-confirm-your-intention-to-delete-data-source'
							),
							[name]
						)}
					</strong>
				</p>
			),
			deleteConfirmationText: sub(Liferay.Language.get('delete-x'), [
				name
			]),
			onClose: close,
			onSubmit: () => {
				deleteDataSource({groupId, id})
					.then(() => {
						addAlert({
							alertType: Alert.Types.Default,
							message: sub(
								Liferay.Language.get(
									'x-is-currently-being-removed-from-analytics-cloud'
								),
								[truncate(name, {length: 50})]
							) as string
						});

						close();

						history.push(
							toRoute(Routes.SETTINGS_DATA_SOURCE_LIST, {
								groupId
							})
						);
					})
					.catch(() => {
						addAlert({
							alertType: Alert.Types.Error,
							message: Liferay.Language.get('error'),
							timeout: false
						});
					});
			},
			title: sub(Liferay.Language.get('delete-x'), [name])
		});
	};

	return (
		<BasePage
			{...otherProps}
			className={getCN('data-source-base-page-root', className)}
			documentTitle={`${
				documentTitle || pageTitle
			} - ${Liferay.Language.get('data-sources')}`}
			groupId={groupId}
			pageActions={
				id && showDelete && currentUser.isAdmin()
					? [
							{
								displayType: 'secondary',
								label: Liferay.Language.get(
									'delete-data-source'
								),
								onClick: handleDeleteClick
							}
					  ]
					: []
			}
			pageDescription={pageDescription || getPageDescription(dataSource)}
			pageTitle={pageTitle}
		>
			<div className='page-container'>
				<div className='content-main'>{passedChildren}</div>

				<div className='content-side'>
					{((dataSource && !hasLegacyDXPConnection(dataSource)) ||
						!dataSource) && (
						<DataSourceStatus
							{...getDataSourceDisplayObject(dataSource)}
						/>
					)}
				</div>
			</div>
		</BasePage>
	);
};

const getOwnChildren = (store, ownProps) => ({
	passedChildren: ownProps.children
});

export default compose<any>(
	withHistory,
	connect(getOwnChildren, {addAlert, close, deleteDataSource, open})
)(BaseDataSourcePage);
