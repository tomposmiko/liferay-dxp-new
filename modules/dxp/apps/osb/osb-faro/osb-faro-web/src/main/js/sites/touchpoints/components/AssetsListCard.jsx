import BasePage from 'shared/components/base-page';
import React from 'react';
import Table from 'shared/components/table';
import TextTruncate from 'shared/components/TextTruncate';
import {Link} from 'react-router-dom';
import {pickBy} from 'lodash';
import {PropTypes} from 'prop-types';
import {toAssetOverviewRoute} from 'shared/util/router';

const CLASSNAME = 'analytics-assets-list';

const ITEMS_SHAPE = PropTypes.shape({
	assetId: PropTypes.string,
	assetType: PropTypes.string,
	interactions: PropTypes.number,
	title: PropTypes.string,
	type: PropTypes.string
});

/**
 * Assets List Card
 * @class
 */
class AssetsListCard extends React.Component {
	static contextType = BasePage.Context;

	static defaultProps = {
		items: []
	};

	static propTypes = {
		items: PropTypes.arrayOf(ITEMS_SHAPE),
		rangeSelectors: PropTypes.object
	};

	constructor(props) {
		super(props);

		this._elementRef = React.createRef();
	}

	/**
	 * Get Asset URL
	 * @description Get url to navigate in an overview
	 * @param {string} assetId
	 * @param {string} assetType
	 * @param {string} title
	 */
	getUrl(assetId, assetType, title) {
		const {
			context: {
				router: {params, query}
			},
			props: {rangeSelectors}
		} = this;

		return toAssetOverviewRoute(
			assetType,
			{
				...params,
				assetId,
				title,
				...(assetType === 'custom' ? {id: assetId} : false)
			},
			pickBy({
				...query,
				rangeEnd: rangeSelectors.rangeEnd,
				rangeKey: rangeSelectors.rangeKey,
				rangeStart: rangeSelectors.rangeStart
			})
		);
	}

	/**
	 * Render Title Column
	 * @param {object} param0
	 */
	renderTitleColumn({assetId, assetType, title}) {
		const url = this.getUrl(assetId, assetType, title);

		return (
			<td className='table-cell-expand'>
				<Link
					className='font-weight-semibold text-truncate-inline text-dark'
					to={url}
				>
					<TextTruncate title={title} />
				</Link>
			</td>
		);
	}

	/**
	 * Render Type Column
	 * @param {object} param0
	 */
	renderTypeColumn({type}) {
		return (
			<td className='table-cell-expand'>
				<div className='font-weight-semibold text-secondary text-truncate-inline'>
					<TextTruncate title={type} />
				</div>
			</td>
		);
	}

	/**
	 * Lifecycle Render - ReactJS
	 */
	render() {
		const {items} = this.props;

		const tableColumns = [
			{
				accessor: 'title',
				cellRenderer: ({data}) => this.renderTitleColumn(data),
				className: 'table-cell-expand',
				label: Liferay.Language.get('asset-name'),
				sortable: false,
				title: true
			},
			{
				accessor: 'type',
				cellRenderer: ({data}) => this.renderTypeColumn(data),
				label: Liferay.Language.get('asset-type'),
				sortable: false
			}
		];

		return (
			<div className={CLASSNAME} ref={this._elementRef}>
				<Table
					className='table-hover'
					columns={tableColumns}
					items={items}
					rowIdentifier={['assetId', 'title']}
				/>
			</div>
		);
	}
}

export default AssetsListCard;
