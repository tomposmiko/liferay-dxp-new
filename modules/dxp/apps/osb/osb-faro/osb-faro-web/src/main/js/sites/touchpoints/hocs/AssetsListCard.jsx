import AssetsListCard from '../components/AssetsListCard';
import AssetsQuery from 'shared/queries/AssetsQuery';
import BaseCard from 'cerebro-shared/components/base-card';
import React from 'react';
import URLConstants from 'shared/util/url-constants';
import {compose} from 'redux';
import {graphql} from '@apollo/react-hoc';
import {HOC_CARD_PROPTYPES} from 'shared/util/proptypes';
import {
	mapPropsToOptions,
	mapResultToProps
} from './mappers/touchpoint-assets-list-query';
import {withEmpty} from 'cerebro-shared/hocs/utils';
import {withError, withLoading} from 'shared/hoc';

const AssetsListWithData = compose(
	graphql(AssetsQuery, {
		options: mapPropsToOptions,
		props: mapResultToProps
	}),
	withLoading({alignCenter: true, page: false}),
	withError({page: false}),
	withEmpty({
		emptyDescription: (
			<>
				<span className='mr-1'>
					{Liferay.Language.get(
						'check-back-later-to-verify-if-data-has-been-received-from-your-data-sources'
					)}
				</span>

				<a
					href={URLConstants.SitesDashboardPagesAssets}
					key='DOCUMENTATION'
					target='_blank'
				>
					{Liferay.Language.get('learn-more-about-display-assets')}
				</a>
			</>
		),
		emptyTitle: Liferay.Language.get(
			'there-are-no-assets-on-the-selected-period'
		)
	})
)(AssetsListCard);

AssetsListWithData.propTypes = HOC_CARD_PROPTYPES;

const defaultProps = {
	className: 'analytics-assets-list-card'
};

const AssetsListBaseCard = ({className, label}) => (
	<BaseCard
		className={className}
		label={label}
		legacyDropdownRangeKey={false}
		minHeight={536}
	>
		{({filters, rangeSelectors, router}) => (
			<AssetsListWithData
				filters={filters}
				rangeSelectors={rangeSelectors}
				router={router}
			/>
		)}
	</BaseCard>
);

AssetsListBaseCard.defaultProps = defaultProps;

export default AssetsListBaseCard;
