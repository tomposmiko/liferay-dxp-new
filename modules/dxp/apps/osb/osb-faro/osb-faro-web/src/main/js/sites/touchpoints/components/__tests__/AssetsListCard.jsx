import AssetsListCard from '../AssetsListCard';
import BasePage from 'shared/components/base-page';
import client from 'shared/apollo/client';
import React from 'react';
import {ApolloProvider} from '@apollo/react-components';
import {BrowserRouter} from 'react-router-dom';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

const items = [
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	},
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	},
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	},
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	},
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	},
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	},
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	},
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	},
	{
		assetId: '01234',
		assetType: 'journal',
		interactions: 1,
		title: '231494203',
		type: 'Web Content'
	}
];

describe('AssetsListCard', () => {
	const MOCK_CONTEXT = {
		router: {
			params: {
				channelId: 123,
				groupId: '2000',
				touchpoint: 'www.liferay.com'
			},
			query: {
				rangeKey: '30'
			}
		}
	};

	const WrappedComponent = props => (
		<ApolloProvider client={client}>
			<BasePage.Context.Provider value={MOCK_CONTEXT}>
				<BrowserRouter>
					<AssetsListCard
						{...props}
						rangeSelectors={{rangeKey: '30'}}
					/>
				</BrowserRouter>
			</BasePage.Context.Provider>
		</ApolloProvider>
	);

	it('should render component', () => {
		const {container} = render(<WrappedComponent />);

		expect(container).toMatchSnapshot();
	});

	it('should render component with items', () => {
		const {container} = render(<WrappedComponent items={items} />);

		expect(container).toMatchSnapshot();
	});
});
