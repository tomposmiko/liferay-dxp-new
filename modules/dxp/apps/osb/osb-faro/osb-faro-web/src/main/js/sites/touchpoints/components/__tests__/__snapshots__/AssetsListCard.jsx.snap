// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetsListCard should render component 1`] = `
<div>
  <div
    class="analytics-assets-list"
  >
    <div
      class="table-responsive table-root flex-grow-1 table-hover"
    >
      <table
        class="table table-autofit table-hover table-heading-nowrap table-nowrap"
      >
        <thead>
          <tr>
            <th
              class="table-head-title table-cell-expand"
            >
              Asset Name
            </th>
            <th
              class="table-head-title"
            >
              Asset Type
            </th>
          </tr>
        </thead>
      </table>
    </div>
  </div>
</div>
`;

exports[`AssetsListCard should render component with items 1`] = `
<div>
  <div
    class="analytics-assets-list"
  >
    <div
      class="table-responsive table-root flex-grow-1 table-hover"
    >
      <table
        class="table table-autofit table-hover table-heading-nowrap table-nowrap"
      >
        <thead>
          <tr>
            <th
              class="table-head-title table-cell-expand"
            >
              Asset Name
            </th>
            <th
              class="table-head-title"
            >
              Asset Type
            </th>
          </tr>
        </thead>
        <tbody
          class="table-hover"
        >
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="table-cell-expand"
            >
              <a
                class="font-weight-semibold text-truncate-inline text-dark"
                href="/workspace/2000/123/assets/web-content/01234/page/www.liferay.com/231494203?rangeKey=30"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  231494203
                </span>
              </a>
            </td>
            <td
              class="table-cell-expand"
            >
              <div
                class="font-weight-semibold text-secondary text-truncate-inline"
              >
                <span
                  class="text-truncate"
                  data-tooltip="false"
                  title=""
                >
                  Web Content
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
